osa_cli_releases/releasing.py

```diff
@@ -225,24 +225,29 @@ def update_ansible_role_requirements_file(filename="", branchname=""):
 
     openstack_roles, external_roles, all_roles = sort_roles(filename)
 
-    if branchname == "master":
-        # TODO(evrardjp): Improve this for ceph
-        for role in external_roles:
-            index = all_roles.index(role)
-            all_roles[index]["version"] = get_sha_from_ref(role["src"], "master")
-
     clone_root_path = tempfile.mkdtemp()
-    for role in openstack_roles:
-        # TODO(evrardjp): use multiprocessing
-        index = all_roles.index(role)
-        version, role_path = clone_role(
-            role["src"], branchname, clone_root_path, depth="1"
-        )
-        # The whole a-r-r is utf-8 encoded, and version is a bytestring.
-        all_roles[index]["version"] = version.decode("utf-8")
-        print("Copying %s's release notes" % role["name"])
-        copy_role_releasenotes(role_path, "./")
-        shutil.rmtree(role_path)
+    for role in all_roles:
+        if role.get("trackbranch"):
+            trackbranch = role["trackbranch"]
+        # This condition can go away when trackbranch merges in master
+        elif branchname == "master":
+            trackbranch = "master"
+        # This condition can go away when trackbranch has merged in
+        # all branches (ocata and above)
+        elif role in openstack_roles:
+            trackbranch = branchname
+        else:
+            trackbranch = None
+        if trackbranch:
```

Comment:
I am wondering if I should add a condition here:
`if trackbranch and trackbranch != 'master'` . Because this will effectively freeze the master branch like the other branches. I like it for stability and consistency, but developers will kill me I think.

Alternatively, I could remove the whole block 232-240, and only track when explicitly asked for. If not asked to track (or `trackbranch: False`), then this code will not change the version anytime (I mean for neither the openstack roles, nor for the external roles), while still regularily copying the release notes. That sounds good enough for me too.

Reply:
Well, I don't think we're quite ready yet to pin roles on master. However, we do want to be able to do a pin to prepare an alpha, beta or rc. So ideally we should be able to tell it to pin master for that purpose, but if the track branch is master and the flag is not set to action the pinning, then it should set the pin to whatever the track branch is - that will make it easy to do the pin for an RC, but then also undo the pin later.

Reply:
There was a discussion for that in the last meeting.

I will split this in two patches: One to handle the freeze, and one for the introduction of trackbranch in a-r-r.